source/elements/oneDAL/source/common_interface/error_handling.rst

```diff
@@ -1,3 +1,79 @@
+.. _error_handling:
+
 ==============
 Error handling
 ==============
+
+Common rules
+============
+
+oneDAL error handling uses exceptions. If an error occurs, it can be propagated
+at the point of a function call. An exception will be thrown and may be caught
+by the user using standard C++ exception handling mechanisms. There are 2 ways
+of handling oneDAL errors. First is through standard C++ exception classes.
+Second is based on our own exceptions hierarchy. Its base class is
+onedal::exception. Exceptions can be divided into subgroups via its inheritors.
+OneDAL exceptions hierarchy is similar to C++ STL exceptions.
+
+Divison rules
+=============
+
+Divison is aligned with C++ STL divison. onedal::exception, onedal::logic_error
+and onedal::runtime_error are abstract and can't be thrown in clear representation.
+Only leaf classes are inherited from the corresponding STL exception classes and could be thrown.
+
+**Inheritance:**
+
+::
+
+    class onedal::exception;
+    class onedal::logic_error : public onedal::exception;
+    class onedal::runtime_error : public onedal::exception;
+
+    class onedal::invalid_argument :  public onedal::logic_error, public std::invalid_argument;
+    class onedal::domain_error :  public onedal::logic_error, public std::domain_error;
+    class onedal::out_of_range :  public onedal::logic_error, public std::out_of_range;
+    class onedal::unimplemented_error :  public onedal::logic_error, public std::logic_error;
+    class onedal::unavailable_error :  public onedal::logic_error, public std::logic_error;
+
+    class onedal::range_error :  public onedal::runtime_error, public std::range_error;
+    class onedal::overflow_error :  public onedal::runtime_error, public std::overflow_error;
+    class onedal::system_error :  public onedal::runtime_error, public std::system_error;
+    class onedal::internal_error :  public onedal::runtime_error, public std::runtime_error;
+
+    class onedal::bad_alloc :  public onedal::exception, public std::bad_alloc;
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Exception
+     - Description
+   * - onedal::exception
+     - It is abstract base class for all throwing exceptions.
+   * - onedal::logic_error
+     - It is abstract class and reports errors that are a consequence of faulty
+       logic within the program such as violating logical preconditions or class
+       invariants and may be preventable.
+   * - onedal::runtime_error
+     - It is abstract class and reports errors that are due to events beyond the scope of the program and can not be easily predicted.
```

Comment:
I think it makes sense to split the table and introduce the separate table for these three as they are abstract